### PR TITLE
Add support for language option

### PR DESCRIPTION
**Pull request by yutannihilation**
See #54

* if `class="language-<lang>"` attribute is available and the `lang` is supported, use it
* if `language="<lang>"` is available and the `lang` is supported, use it (I'm not sure where I found this option, though...)
* otherwise use `text`

By default, conflr allows only these languages:

* SQL (`sql`)
* Rcpp (`cpp`)
* Python (`py`)
* HTML (`xml`)
* CSS (`css`)
* Bash (`bash`)
* YAML (`yaml`)

If the user need to allow other languages (e.g. R), the option below needs to be set.

``` r
options(conflr_supported_syntax_highlighting = "r")
```

or in the named form

``` r
options(conflr_supported_syntax_highlighting = c(<markdown language name> = "<confluence macro name>"))
```

### CODE REVIEW

File: R/translate.R [hunk 1]
```diff
@@ -63,17 +63,71 @@ restore_cdata <- function(x) {
   x
 }
 
+# picking some popular languages from https://confluence.atlassian.com/doc/code-block-macro-139390.html
+supported_languages_default <- c(
```
Comment (cderv) [inline R/translate.R]: Why putting only three of them ? 
I understand Rmarkdown won't be used for all those language though... 
I would add in the default **Bash**, **CSS**, **HTML**, **JavaScript**, **YAML**
I already used those ones in Rmd document, mainly for documentation purposes. 

The extra supported languages mechanism will allow to add those, so feel free to keep the defaut you think are best.

Comment (yutannihilation) [reply R/translate.R]: Thanks, agreed!

**Comment by yutannihilation** [comment]
Hmm, it seems `language` options are different between versions...

[v5.9](https://confluence.atlassian.com/conf59/code-block-macro-792499083.html#CodeBlockMacro-Storageformatexample) uses "html/xml" for HTML and XML, but our Confluence returns "xml" (I'm not sure about the version).
https://confluence.atlassian.com/doc/code-block-macro-139390.html

**Comment by yutannihilation** [comment]
`v5.9` seems too old, so let's use "xml".

**Comment by cderv** [comment]
FWIW, we have 6.13.8

**Comment by yutannihilation** [comment]
@cderv 
(Just for notice) Though this PR is still work in progress (especially about how the option is exposed to the users), I'm considering merging this so that I can refactor the internals. Even if I choose to do so, I'll keep #54 open, so if you have some feedbacks please comment there (or here).

**Comment by yutannihilation** [comment]
Thanks, yeah. I'll call you when it's ready.

**Pull request merged**